backend/enums/activity.js

```diff
@@ -102,4 +123,45 @@ export default class Activity {
     )
     return feeds.sort((a, b) => (a.timestamp < b.timestamp ? 1 : -1))
   }
+
+  static async convertToFeedActivity(activity, postId, authorId, postTime) {
+    const post = await models.posts.findByPk(postId)
+    const postDto = await PostDTO.convertToDto(post)
+    return {
+      post: postDto,
+      timestamp: Date.parse(postTime),
+      activity: activity.type,
+      userId: authorId,
+    }
+  }
+
+  static async getPostsForInterests(userId) {
+    const posts = []
+    // Retrieve user's interests
+    const interestsDB = await models.userInterests.findAll({
+      where: {
+        userID: userId,
+      },
+    })
+
+    const relatedPosts = []
+
+    // Retrieve post IDs relevant to the user's interests
+    for (let interest of interestsDB) {
+      relatedPosts.push(
+        ...(await models.postTag.findAll({
+          where: {
+            tagID: interest.tagId,
+          },
+        }))
+      )
+    }
```

Comment:
```suggestion
    const postsInterest = await Promise.all(interestsDB.map(interest => (
       await models.postTag.findAll({
          where: {
            tagID: interest.tagId,
          },
       }))
     )
     
     const relatedPosts = postsInterest.reduce((posts, interests) => [...posts, ...interests], [])
```

recommend using `Promise.All()` to call it concurrently, currently, it need to await each element before fetching next element. Using reduce to flatten the array from [[],[],[]] format to []